lib/web_apis/browser.es6.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+foam.CLASS({
+  name: 'Browser',
+  package: 'com.web.api',
+  properties: [
+    {
+      name: 'browserName',
+      required: true,
+    },
+    {
+      name: 'browserVersion',
+      required: true,
+    },
+    {
+      name: 'OS',
+      required: true,
+    },
+    {
+      name: 'OSVersion',
+      required: true,
+    },
+    {
+      name: 'id',
+    },
+  ],
+  methods: [
+    {
+      name: 'init',
+      document: `After the browser object is created, hashCode function is
+        used to assign this browser a unique browser id`,
+      code: function() {
+        this.id = foam.util.hashCode(this);
+      },
+    },
+    {
+      name: 'getBrowserKey',
```

Comment:
If you're calling this much, you could make it a Property with an expression, in which case you would get caching for free. Not a big deal though if you don't call much.